src/codegen/generators/expression_generator.rs

```diff
@@ -250,15 +250,58 @@ impl<'a, 'b> ExpressionCodeGenerator<'a, 'b> {
         //Generate a load for the qualifer
         let (expression_type, value) = self.generate_expression(&expression)?;
         if let AstStatement::DirectAccess { access, index, .. } = last {
+            let datatype = self
+                .annotations
+                .get_type_or_void(last, self.index)
+                .clone_type_information();
             //Generate and load the index value
-            let index = access.to_bits(*index);
-            let datatype = self.annotations.get_type_or_void(last, self.index);
-            let rhs = self
-                .llvm_index
-                .get_associated_type(expression_type.get_name())
-                .unwrap()
-                .into_int_type()
-                .const_int(index.into(), false);
+            let rhs = match &**index {
+                AstStatement::LiteralInteger { value, .. } => {
+                    //Convert into the target literal
+                    let bitwidth = access.get_bit_witdh();
+                    let value: u64 = (*value).try_into().unwrap_or_default();
```

Comment:
You assume that you will never see a negative number?

Reply:
It's currently not possible at the parser level. See my comment there.